core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -338,6 +340,9 @@ private[spark] class Executor(
             metricsSystem = env.metricsSystem)
           threwException = false
           res
+        } catch {
+          case _: CommitDeniedException =>
+            throw new TaskKilledException("commit denied")
```

Comment:
Doesn't a stage abort also cause tasks to show up as killed (due to "stage cancelled"?) https://github.com/apache/spark/blob/95aef660b73ec931e746d1ec8ae7848762ba0d7c/core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala#L1531

It seems to me that CommitDenied always implies the task is killed, in which case it might be fine to convert all CommitDeniedExceptions into TaskKilled.

Btw, there's a catch block below -- `case CausedBy(cDE: CommitDeniedException) =>` which seems like the right place to be doing this handling.